### PR TITLE
[OCPBUGS#163] Install config initialization conditional stack mystery

### PR DESCRIPTION
**Pull request by maxwelldb**
Version(s):
4.6+, in various capacities

Issue:
[OCPBUGS-163](https://issues.redhat.com/browse/OCPBUGS-163)

Link to docs previews: 
- [openshift-enterprise](http://file.rdu.redhat.com/~mbridges/conditional-mystery/welcome/)
- [openshift-origin](http://file.rdu.redhat.com/~mbridges/conditional-mystery-origin/welcome/)

_Note:_ You must use a portal build to catch these particular errors in context. 

The problem does look like it's resolved on the page mentioned in the bug if I build from this branch.

---
 I kludge-scripted a look across all modules for offending items, here. The problem doesn't seem to be from any of the OSP modules--rather, there is an unmatched `restricted` attribute in a page that appears before the ShiftStack docs. This issue only occurs on the portal, as portal builds rely on a single `master.adoc` file. 
 
 Resolving these `:restricted:` mismatches might help: 

- [x]  installation-network-user-infra.adoc
- [x] installation-aws-user-infra-installation.adoc
- [x] installation-load-balancing-user-infra.adoc
- [x] cli-installing-cli-web-console-macos.adoc
- [x] installation-vsphere-config-yaml.adoc
- [x] installation-user-infra-machines-static-network.adoc
- [x] installation-complete-user-infra.adoc

Aside from this, I'd like to verify that the openshift-origin _closing only_ items were not added as gum + tape for this bug. As the original writer is no longer documenting this content, I'll need to verify with a currently responsible party that this change wouldn't ruin anything.

### CODE REVIEW

**Comment by maxwelldb** [comment]
I think I kludge scripted a look across all modules for offending items, here. The problem doesn't seem to be from any of the OSP modules--rather, there is an unmatched `restricted` attribute in a page that appears before the ShiftStack docs. 
 
 Resolving these mismatches might help. 
 

- [ ]  installation-network-user-infra.adoc
- [ ] installation-aws-user-infra-installation.adoc
- [ ] installation-load-balancing-user-infra.adoc
- [ ] cli-installing-cli-web-console-macos.adoc
- [ ] installation-vsphere-config-yaml.adoc
- [ ] installation-user-infra-machines-static-network.adoc
- [ ] installation-complete-user-infra.adoc

Aside from this, I'd like to verify that the openshift-origin _closing only_ items were not added as gum + tape for this bug. 

TODO:
- Verify openshift-origin changes with original writer

**Comment by maxwelldb** [comment]
@SNiemann15 @pneedle-rh @xenolinux Could you take a look at this at your leisure? This PR impacts changes that you made in the past and/or files that you updated recently.

**Comment by pneedle-rh** [comment]
@maxwelldb hi! Thank you for looking into this. I realise that this is a difficult task, given that there are so many context-specific temporary attributes set and unset in the module files in the "Installing" book. I think a review is equally difficult to complete holistically.

### Customer Portal build previews

I cloned your feature branch locally, ran the `build_for_portal.py` script and built certain pages in the `drupal-build/openshift-enterprise/installing/` directory using Asciidoctor. From a review of those previews, I can't see content that is specific to the restricted network use case in the non-restricted pages, but it's difficult to review all of them. The restricted content in the `installing/installing_openstack/installing-openstack-user.html` page certainly seems to have been removed successfully, as you mentioned.

### Reviewing the temporary attributes in the "Installing" book

I was thinking about how I could help with the more detailed review of the temporary attribute settings. I used the [list-adoc-files-in-a-distro.py](https://gitlab.cee.redhat.com/pneedle/pneedle-scripts/-/blob/main/osdocs_scripts/list-adoc-files-in-a-distro.py) script to generate a list of assemblies and modules that are actively live in the "Installing" book in the `main` branch. This is in a clone that uses the latest mainstream content without including the changes in your PR. I did this by manually removing all but the "Installing" book entries in `_topic_maps/_topic_map.yaml` in a test branch built from `main` and running the script for the `openshift-enterprise` distro. I will attach the file that contains the active module list (`eval-check-openshift-enterprise-active-module-list.txt`) in a separate comment.

I ran two commands. The first printed the `ifeval` statements in each of those module files:

`$ for i in $(cat eval-check-openshift-enterprise-active-module-list.txt); do echo "#### $i:"; sed -n -e '/ifeval/,/endif/ p'  /var/tmp/clones/pneedle/openshift-docs/$i; echo; done > /<dir>/pr49742/evals-in-install-book-main.txt`

The second removed the `ifeval` and `endif` lines from that output, to make it easier to review which ones are closed in the same module:

`$ for i in $(cat eval-check-openshift-enterprise-active-module-list.txt); do echo "#### $i:"; sed -n -e '/ifeval/,/endif/ p'  /var/tmp/clones/pneedle/openshift-docs/$i; echo; done | grep -v 'ifeval\|endif' > /<dir>/pr49742/evals-in-install-book-main-attributes-only.txt`

This gives us a view of the attributes set context to each `ifeval` statement and a way to scan for potential anomalies. I also manually created another file that highlights the modules that might have anomalies in them, from my own manual scan of those files. I will upload these files to this PR separately:

- `evals-in-install-book-main.txt`
- `evals-in-install-book-main-attributes-only.txt`
- `evals-in-install-book-main-that-do-not-seem-to-match.txt`

Here is the list of files highlighted in the last of those files. My apologies if I overlooked any:

```
$ grep -is '^###' /home/pneedle/data/documents/technical_writing/pr49742/evals-in-install-book-main-that-do-not-seem-to-match.txt
#### modules/installation-about-custom-azure-vnet.adoc:
#### modules/installation-complete-user-infra.adoc:
#### modules/installation-configuration-parameters.adoc:
#### modules/installation-launching-installer.adoc:
#### modules/installation-load-balancing-user-infra.adoc:
#### modules/installation-network-user-infra.adoc:
#### modules/installation-supported-aws-machine-types.adoc:
#### modules/installation-user-infra-generate-k8s-manifest-ignition.adoc: (One to check because the order is different in the cancellations)
#### modules/installation-user-infra-machines-static-network.adoc:
#### modules/machineset-azure-enabling-accelerated-networking-new-install.adoc:
#### modules/nw-operator-cr.adoc:
```

It's difficult to run a similar exercise for contexts set in `ifdef` and `ifndef` statements, given that they might be nested. Trying another approach, I listed all of the attribute settings in each of the active modules in the Installing book without the context of `ifeval`, `ifdef` and `ifndef` statements:

----
$ for i in $(cat eval-check-openshift-enterprise-active-module-list.txt); do echo "#### $i:"; grep -is '^:' /var/tmp/clones/pneedle/openshift-docs/$i | grep -v '_content-type'; echo; done > /<dir>/pr49742/attribute-settings-in-install-book-main.txt
----

I will also upload the `attribute-settings-in-install-book-main.txt` file separately.

Although this doesn't provide the detailed context for the temporary attribute settings in each module, it does provide another way to highlight which active module files might not unset their temporary attributes.

I hope this helps. Please let me know if you have any questions.

**Comment by pneedle-rh** [comment]
I also just checked and there are a few attribute files that are active in the "Installing" book in the openshift-enterprise distro that have some attribute settings that might need to be unset. I've uploaded a file that summarises all of the attribute settings for those active modules, to make any potential anomalies easier to scan:

- [attribute-settings-in-install-book-assemblies-main.txt](https://github.com/openshift/openshift-docs/files/9505471/attribute-settings-in-install-book-assemblies-main.txt)

**Comment by maxwelldb** [comment]
@pneedle-rh Looks quite useful! Thanks for the work.

File: modules/installation-user-infra-machines-static-network.adoc
Comment (SNiemann15) [inline modules/installation-user-infra-machines-static-network.adoc:543]: @alishaIBM I did a quick check of the Power docs and noticed this file doesn't show in your assemblies at all (both in the preview and the published docs) . Is that on purpose?

**Review by SNiemann15 — APPROVED** [review]
@maxwelldb IBM Z install assemblies still look good!

@alishaIBM Can you please verify for IBM Power.

File: modules/installation-user-infra-machines-static-network.adoc
Comment (maxwelldb) [reply modules/installation-user-infra-machines-static-network.adoc:543]: @alishaIBM Any update here?

**Comment by maxwelldb** [comment]
I intend to address non-`:restricted:`, non-installation issues in another PR. While I won't merge until the IBM Power content is okayed, advancing to peer review.

**Review by adellape — APPROVED** [review]

File: modules/installation-user-infra-machines-static-network.adoc
Comment (SNiemann15) [reply modules/installation-user-infra-machines-static-network.adoc:543]: @maxwelldb I went through the Power docs with @alishaIBM today and we found the content of that file in the preview. I only missed it because it didn't show in the leftnav toc. Reason is they used `leveloffset=+3` and not 2 like bare metal/IBM Z .  Everything else also looks good for the Power assemblies.

Comment (alishaIBM) [reply modules/installation-user-infra-machines-static-network.adoc:543]: @maxwelldb the IBM Power install assemblies look good. Thanks.

Comment (maxwelldb) [reply modules/installation-user-infra-machines-static-network.adoc:543]: Thanks! @SNiemann15 @alishaIBM I'll merge tomorrow unless I hear anything else.

**Review by alishaIBM — APPROVED** [review]

**Pull request merged**
